src/simplify_algebra.cpp

```diff
@@ -1237,11 +1237,11 @@ struct find_neg_unit_ops
     auto matcher() const
     {
         auto mul_neg_1 = match::name("mul")(
-            match::either_arg(0, 1)(match::has_value(-1.0f), match::any().bind("x")));
+            match::either_arg(0, 1)(match::has_value(-1.0f, 1, 0), match::any().bind("x")));
         auto div_neg_1 =
-            match::name("div")(match::args(match::any().bind("x"), match::has_value(-1.0f)));
-        auto sub_0 =
-            match::name("sub")(match::args(match::has_value(0.0f), match::any().bind("x")));
+            match::name("div")(match::args(match::any().bind("x"), match::has_value(-1.0f, 1, 0)));
+        auto sub_0 = match::name("sub")(
+            match::args(match::has_value(0.0f, 1e-3, 0), match::any().bind("x")));
```

Comment:
`atol_mul` should be 0. Also there is no need to set rtol.

Reply:
That works too.